compiler/circle-execution-plan/src/ExecutionPlanner.cpp

```diff
@@ -86,18 +86,195 @@ void create_allocation_node(Json::Value &allocations_node,
   allocations_node.append(allocation_node);
 }
 
+// TODO: Introduce inplace optimization
+bool can_be_inplace_optimization_node(luci::CircleNode *node)
+{
+  switch (node->opcode())
+  {
+    case luci::CircleOpcode::LOGISTIC:
+    case luci::CircleOpcode::RESHAPE:
+    case luci::CircleOpcode::EXPAND_DIMS:
+      return true;
+    default:
+      return false;
+  }
+}
+
 } // namespace
 
-void ExecutionPlanner::make_execution_plan()
+void ExecutionPlanner::make_execution_plan_luci_interpreter()
 {
   get_default_execution_order_plan();
   _required_size = get_offsets_with_greedy_by_size();
+
   for (uint32_t i = 0; i < _ordered_nodes.size(); i++)
   {
     luci::CircleNodeExecutionPlan execution_plan(i, _offsets[i]);
     luci::add_execution_plan(loco::must_cast<luci::CircleNode *>(_ordered_nodes[i]),
                              execution_plan);
   }
+
+  printf("Buffer required memory = %d\n", _required_size);
+  dump_inform();
+}
+
+void ExecutionPlanner::make_execution_plan_onert_micro_base()
+{
+  switch (_buffers_type)
+  {
+    case SupportedBuffersType::COMMON:
+      make_execution_plan_onert_micro_common_buffer();
+      break;
+    case SupportedBuffersType::SPLIT:
+      make_execution_plan_onert_micro_split_buffer();
+      break;
+    default:
+      throw std::runtime_error("Unsupported buffer type\n");
+  }
+}
+
+void ExecutionPlanner::make_execution_plan_onert_micro_split_buffer()
+{
+  const auto input_size = _graph->inputs()->size();
+  const auto output_size = _graph->outputs()->size();
+
+  // Make execution plan for inputs
+  _ordered_nodes = loco::input_nodes(_graph);
+  const auto input_required_size = get_offsets_with_greedy_by_size();
+
+  for (uint32_t i = 0; i < _ordered_nodes.size(); i++)
+  {
+    luci::CircleNodeExecutionPlan execution_plan(i, _offsets[i]);
+    luci::add_execution_plan(loco::must_cast<luci::CircleNode *>(_ordered_nodes[i]),
+                             execution_plan);
+  }
+  dump_inform();
+  printf("Input graph buffer required memory = %d\n", input_required_size);
```

Comment:
Maybe move this (and output/intermediate tensor) code to a separate function?

Reply:
done, please see https://github.com/Samsung/ONE/pull/9893